src/ngraph/runtime/cpu/mkldnn_emitter.cpp

```diff
@@ -114,12 +114,14 @@ mkldnn::memory::desc
 
 size_t MKLDNNEmitter::build_memory_primitive(const mkldnn::memory::desc& desc)
 {
-    // The MKL-DNN C++ API forces proper initialization of a memory primitive
-    // with a non-null pointer (unlike the C API)
-    // Primitives are initialized at runtime so we use a known-invalid address here
-    // to bypass this check
-    return insert_primitive(
-        new mkldnn::memory({desc, mkldnn_utils::global_cpu_engine}, reinterpret_cast<void*>(0x42)));
+    // MKLDNN unconditionally writes into data handles at primitive creation for padded layouts
+    // Provide a dummy location to keep MKLDNN happy
+    auto pd = mkldnn::memory::primitive_desc(desc, mkldnn_utils::global_cpu_engine);
+    void* tmp_buf = malloc(pd.get_size());
```

Comment:
malloc can return null. Are we okay with mkldnn segfaulting instead of us throwing an exception?

Reply:
Good point. Will add a check.